### DISCUSSION TITLE
Pas d'auto-discovery HA

### DISCUSSION THREAD

**Discussion by laurentftech** [MyElectricalData/myelectricaldata_import#465] [Q&A]
Bonjour. 
J'ai installé myelectricaldata avec Docker. J'arrive bien à faire remonter les données dans Energy de Homeassistant (sur Docker aussi) par contre je n'arrive pas à faire fonctionne l'auto-discovery.
Il n'est pas clair pour moi s'il faut créer un serveur MQTT ?
Sinon, je ne comprends pas trop où est le problème (problème de communication entre containers ?).
Merci pour votre aide.

Ma configuration MED est la suivante:

```
cycle: 14400  # 4H
debug: true
log2file: false
wipe_influxdb: false
tempo:
  enable: true
  price_blue_hc: 0.1464
  price_blue_hp: 0.2460
  price_red_hc: 0.2460
  price_red_hp: 0.7324
  price_white_hc: 0.1464
  price_white_hp: 0.2460
home_assistant:
  enable: true
  discovery: true
  discovery_prefix: homeassistant
home_assistant_ws:
  enable: true
  ssl: true
  token: ffhff
  url: homeassistant.mydomain.fr
influxdb:
  enable: true
  hostname: influxdb
  port: 8086
  token: yyy
  org: Personal
  bucket: MyElectricalData
  method: batching
  batching_options:
    batch_size: 1000
    flush_interval: 1000
    jitter_interval: 0
    retry_interval: 5000
    max_retry_time: 180_000
    max_retries: 5
    max_retry_delay: 125_000
    exponential_base: 2
mqtt:
  enable: false

myelectricaldata:
  "yyy":
    enable: 'true'
    token: xxxx
    name: "Maison"
    addresses: 'true'
    cache: 'true'
    consumption: 'true'
    consumption_detail: 'true'
    consumption_price_base: '0.228'
    consumption_price_hc: "0.1464"
    consumption_price_hp: "0.2460"
    consumption_max_date: "2021-06-01"
    consumption_detail_max_date: "2021-06-01"
    offpeak_hours_0: 20H00-8H00;13H00-18H00   # LUNDI
    offpeak_hours_1: 20H00-8H00;13H00-18H00   # MARDI
    offpeak_hours_2: 20H00-8H00;13H00-18H00   # MERCREDI
    offpeak_hours_3: 20H00-8H00;13H00-18H00   # JEUDI
    offpeak_hours_4: 20H00-8H00;13H00-18H00   # VENDREDI
    offpeak_hours_5: 22H00-6H00;12H00-14H00   # SAMEDI
    offpeak_hours_6: 20H00-8H00;13H00-18H00   # DIMANCHE
    production: 'false'
    production_detail: 'false'
    production_price: '0.0'
    production_max_date: "2021-06-01"
    production_detail_max_date: "2021-06-01"
    refresh_addresse: 'false'
    refresh_contract: 'false'
```

Est-il indispensable d'utiliser le port 5000 pour MED ? (je publie actuellement sur un autre port car le 5000 est utilisé par le synology)

**Comment by m4dm4rtig4n** [comment]
Hello, yep MQTT est obligatoire pour l'auto-discovery des sensor dans Home Assistant .

Et dans le fichier de config tu peut changer le port si besoin : 
port: XXXX

Je viens de voir que je n'ai pas update la doc à ce sujet d'ailleurs :)

**Reply by laurentftech** [reply]
Merci. Bon j'ai réussi à le connecter à HA via l'intégration MQTT, pas réussi à faire fonctionner l'auto-discovery. 
La description n'est pas très claire pour les nouveaux:

"La dépendance à MQTT n'est plus obligatoire et je supporte :

MQTT
Home Assistant (Via MQTT Auto Discovery)
InfluxDB"

Ça donne l'impression qu'il n'y a plus besoin d'avoir un serveur MQTT externe pour se connecter à Home Assistant.

**Reply by m4dm4rtig4n** [reply]
C'est parce qu'avant le service ne démarrer pas sans service MQTT.
Que maintenant c'est un choix, tu peut démarrer MED et exporter uniquement dans un influxdb si tu le souhaitre par exemple.
